knimin/lib/tests/test_data_access.py

```diff
@@ -500,200 +501,180 @@ def test_delete_study(self):
+        db.write_sample_plate_layout(plate_id, layout)
+        samples.remove('9999.sample2')
+
+        with self.assertRaises(ValueError) as ctx:
+            db.delete_study(9999)
+        self.assertEqual(ctx.exception.message,
+                         "Can't remove study 9999, samples have been plated")
+
+        # Try to add samples to a study that does not exist
+        with self.assertRaises(ValueError) as ctx:
+            db.set_study_samples(9998, ['9998.sample1', '9998.sample2'])
+        self.assertEqual(ctx.exception.message,
+                         'Study ID 9998 does not exist.')
+
+    def test_get_study_plated_samples(self):
+        # Create a study
+        db.create_study(9999, title='LabAdmin test project', alias='LTP',
+                        jira_id='KL9999')
+        self._clean_up_funcs.append(partial(db.delete_study, 9999))
+
+        # Retrieve plated samples from an empty study
+        self.assertEqual(db.get_study_plated_samples(9999), {})
+
+        # Retrieve plated samples from a study with no samples plated
+        db.set_study_samples(
+            9999, ['9999.sample1', '9999.sample2', '9999.sample3'])
+        self.assertEqual(db.get_study_plated_samples(9999), {})
+
+        # Put samples in a couple of plates
+        # Create a plate
+        pt = db.get_plate_types()[0]
+        plate_id = db.create_sample_plate('Test plate', pt['id'],
+                                          'test', [9999])
+        self._clean_up_funcs.insert(
+            0, partial(db.delete_sample_plate, plate_id))
+        plate_id_2 = db.create_sample_plate('Test plate 2', pt['id'],
+                                            'test', [9999])
+        self._clean_up_funcs.insert(
+            0, partial(db.delete_sample_plate, plate_id_2))
+        # Create the layout
+        layout = []
+        row = []
+        for i in range(pt['rows']):
+            for j in range(pt['cols']):
+                row.append({'sample_id': None, 'name': None, 'notes': None})
+            layout.append(row)
+            row = []
+        layout[0][0]['sample_id'] = '9999.sample1'
+        layout[0][1]['sample_id'] = '9999.sample2'
+        db.write_sample_plate_layout(plate_id, layout)
+        layout[0][1]['sample_id'] = '9999.sample3'
+        db.write_sample_plate_layout(plate_id_2, layout)
+
+        exp = {plate_id: ['9999.sample1', '9999.sample2'],
+               plate_id_2: ['9999.sample1', '9999.sample3']}
+        self.assertEqual(db.get_study_plated_samples(9999), exp)
+
+        # Try to get plated samples from a study that does not exist
+        with self.assertRaises(ValueError) as ctx:
+            db.get_study_plated_samples(9998)
+        self.assertEqual(ctx.exception.message,
+                         'Study ID 9998 does not exist.')
+
+    def test_get_study_samples(self):
+        # Create a study
+        db.create_study(9999, title='LabAdmin test project', alias='LTP',
```

Comment:
just remember that the jira_id is not the same than the jira_key ... but just double checked the documentation and you can instantiate a jira project either by key or id ...